modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -969,8 +969,16 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
             int stripesPerSample;
             size_t stripeSize;
             Range r = r0;
-
-            if( nstripes >= batchSize*2 )
+            bool depthWiseConvolution = !is1x1 && isConv2D && ngroups > 1 && batchSize == ngroups && inpCn == 1 &&
```

Comment:
why we check `batchSize == ngroups` and `ngroups > 1`? depthwise convolution is a convolution layer with groups == input channels == output channels. So the question is what we expect to match `depthWiseConvolution`?

Reply:
@dkurt, thank you! I think, you are right. Since `inpCn` and `outCn` are computed as `input_->size[1]/ngroups` and `output->size[1]/ngroups`, respectively, the condition should be `ngroups > 1 && inpCn == 1 && outCn == 1`, where the latter two parts are already there. So we should just remove `batchSize == ngroups` check. I'm now improving the patch, will put in this change as well.